rtl/eclrtl/eclregex.cpp

```diff
@@ -183,19 +191,37 @@ class CCompiledStrRegExpr : implements ICompiledStrRegExpr
 {
 private:
     pcre2_code_8 * compiledRegex = nullptr;
+    bool isUTF8Enabled = false;
 
 public:
-    CCompiledStrRegExpr(const char * _regex, bool _isCaseSensitive = false)
+    CCompiledStrRegExpr(const char * _regex, bool _isCaseSensitive, bool _enableUTF8)
+    : isUTF8Enabled(_enableUTF8)
     {
         int errNum = 0;
         PCRE2_SIZE errOffset;
-        uint32_t options = (_isCaseSensitive ? 0 : PCRE2_CASELESS);
+        uint32_t options = ((_isCaseSensitive ? 0 : PCRE2_CASELESS) | (_enableUTF8 ? PCRE2_UTF : 0));
 
         compiledRegex = pcre2_compile_8((PCRE2_SPTR8)_regex, PCRE2_ZERO_TERMINATED, options, &errNum, &errOffset, pcre2CompileContext8);
 
         if (compiledRegex == nullptr)
         {
-            failWithPCRE2Error(errNum, "Error in regex pattern: ", _regex);
+            failWithPCRE2Error(errNum, "Error in regex pattern: ", _regex, errOffset);
+        }
+    }
+
+    CCompiledStrRegExpr(int _regexLength, const char * _regex, bool _isCaseSensitive, bool _enableUTF8)
+    : isUTF8Enabled(_enableUTF8)
+    {
+        int errNum = 0;
+        PCRE2_SIZE errOffset;
+        uint32_t options = ((_isCaseSensitive ? 0 : PCRE2_CASELESS) | (_enableUTF8 ? PCRE2_UTF : 0));
+        size32_t regexLength = (isUTF8Enabled ? rtlUtf8Size(_regexLength, _regex) : _regexLength);
```

Comment:
regexSize a better variable name for consistency

Reply:
Agreed; changed.